File: RestAPI/Prodaja karata za gradski prijevoz/Presentation/DTO/Admin/Vehicles/VehicleDto.cs
using System.ComponentModel.DataAnnotations;

namespace Presentation.DTO.Admin.Vehicles
{
    public class VehicleDto
    {
        [Required]
        public int Number { get; set; }
        [Required]
        public string RegistrationNumber { get; set; } = null!;
        [Required]
        public string Color { get; set; } = null!;
        [Required]
        public Guid ManufacturerId { get; set; }
        [Required]
        public Guid VehicleTypeId { get; set; }
        [Required]
        public int BuildYear { get; set; }
        [Required]
        public Guid VehicleId { get; set; }
    }
}

File: SPA/src/app/models/Admin/Vehicle/VehicleDto.ts
import { ManufacturerDto } from "./manufacturerDto";
import { VehicleTypeDto } from "./vehicleTypeDto";

interface VehicleDtoBase {
  id?: string,
  number?: number;
  registrationNumber?: string;
  color?: string;
  buildYear?: number;
}

export interface VehicleDto extends VehicleDtoBase {
  manufacturerId?: string;
  vehicleTypeId?: string;
}

export interface VehicleListDto extends VehicleDtoBase {
  manufacturer: ManufacturerDto;
  type: VehicleTypeDto;
  hasMalfunction: boolean;
}
